### ISSUE TITLE
[V3] Implement new Concept Exercise: errors

### ISSUE THREAD

**Issue by SleeplessByte** [exercism/javascript#949]
This issue describes how to implement the `errors` concept exercise for the `javascript` track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

See the documentation _above_ (general documentation), as well as [How to implement a Concept Exercise in JavaScript](https://github.com/exercism/v3/blob/master/languages/javascript/reference/implementing-a-concept-exercise.md).

## Goal

The goal of this exercise is to teach the student how to handle errors / exception, throw them and create their own.

## Learning objectives

- `try {} catch {}`
- `throw Error`
- `instanceOf SpecialError`
- Creating custom errors
  - Prefilling a message 
  - Constructing a message based on some constructor value
  - Storing constructor values on the error (so that you have context)

## Out of scope

- Promise rejection
- Throwing non-errors

## Concepts

- `errors`

## Prerequisites

- `strings`
- `classes-intro`

## After

- Throwing non-errors
- Stacktrace
- [Actionable errors](https://pasteapp.com/p/6y4mifP7itH/s/nAmDoqfNK0H?view=FwpyUzWlnqi)

## Help

You can choose to do this solo-style, or collaborate with multiple people on this. My suggestion would be is to 

1. **first accept this issue by saying "I'd like to work on this"** (you will not get a response from me. Just go with it) and optionally request that someone works with you (and wait for a second person to accept your request).
2. Use this issue to write down what you need (collect reference material) and discuss, if necessary, what to write and what not to write.
3. Draft a PR and request the feedback from at least one other JavaScript member (I recommend just pinging the channel on slack).
